src/framework/core/js/ResourceLoader.js

```diff
@@ -13,59 +13,672 @@ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
+     * URLs within the specification.
+     * @return {ResourceFetcher} The ResourceFetcher ready to be attached to the ResourceLoader's top level
+     */
+    fluid.resourceLoader.makeResourceFetcher = function (that, resourceOptions, resolveResources) {
+        var resolved = resolveResources();
+        var fetcher = fluid.makeResourceFetcher(fluid.copy(resolved), null, resourceOptions);
+        // Note that we beat the existing completion listener in the fetcher by "sheer luck"
+        fluid.each(fetcher.resourceSpecs, function (resourceSpec, key) {
+            resourceSpec.promise.then(function () {
+                that.resources[key] = resourceSpec;
+            }, function (err) {
+                that.events.onResourceError.fire(err);
+            });
+        });
+        return fetcher;
+    };
+
+    // TODO: This function needs to be eliminated and transformURL moved into the body of makeResourceFetcher next
+    // to explodeForLocales, where it can make use of the `pathKey`. Then we can move the obstrusive `terms` definition
+    // out of top level. Unfortunately it is used explicitly within the fluid.prefs.separatedPanel.lazyLoad
+    // which we need to unpick since its workflow can now probably be substantially simplified.
+    fluid.resourceLoader.resolveResources = function (resources, locale, defaultLocale, resourceOptions, transformURL) {
+        return fluid.transform(resources, function (record) {
+            var userSpec = typeof(record) === "string" ? {url: record} : record;
+            var resourceSpec = $.extend(true, {}, resourceOptions, {
+                defaultLocale: defaultLocale,
+                locale: locale}, userSpec);
+            resourceSpec.url = transformURL(resourceSpec.url);
+            return resourceSpec;
+        });
+    };
 
-        return fluid.transform(mapped, function (url) {
-            var resourceSpec = {url: url, forceCache: true, options: that.options.resourceOptions};
-            return $.extend(resourceSpec, fluid.filterKeys(that.options, ["defaultLocale", "locale"]));
+    /** On construction of the resourceLoader, kick off the process of fetching all the resources configured within
+     * its resourceFetcher. Note that some or all of these resources may already have been fetched by demands occuring
+     * during component startup (e.g. as initial model values or renderer templates), and so the `onResourcesLoaded`
+     * event may fire immediately
+     * @param {fluid.resourceLoader} that - The loader for which the I/O fetch process is to be started
+     */
+    fluid.resourceLoader.loadResources = function (that) {
+        var completionPromise = that.resourceFetcher.fetchAll();
+        completionPromise.then(function () {
+            that.events.onResourcesLoaded.fire(that.resources);
+        }, function (error) {
+            // Note that if the failure was for a resource demanded during startup, this component will already have
```

Comment:
Shouldn't you be calling onResourceError here?

Reply:
It will already have been fired by the individual resource failure on line 640